src/telemetry/metrics/labels.rs

```diff
@@ -244,11 +214,11 @@ impl fmt::Display for Classification {
 
 // ===== impl Direction =====
 
-impl Direction {
-    fn from_context(context: &ctx::Proxy) -> Self {
-        match context {
-            &ctx::Proxy::Inbound(_) => Direction::Inbound,
-            &ctx::Proxy::Outbound(_) => Direction::Outbound,
+impl<'a> From<&'a ctx::Proxy> for Direction {
```

Comment:
FWIW, I had originally written this as a `From` conversion when I first added the `Direction` enum, but @briansmith asked me to change it: https://github.com/linkerd/linkerd2/pull/661#discussion_r179904788. Personally, I like it better as a `From` conversion, but I figured I ought to put that out there...